### DISCUSSION TITLE
can't reset the nrf51822 chip

### DISCUSSION THREAD

**Discussion by shang-huan** [bitcraze/discussions#493] [Q&A]
When I try to reset the nrf51822 chip, It can't work. The execution results are as follows:
ou@outaotaodeMacBook-Air crazyflie2-nrf-firmware % make factory_reset_21
make mass_erase
openocd -d2 -f interface/stlink.cfg  -f target/nrf51.cfg -c init -c targets -c "reset halt" \
                 -c "nrf51 mass_erase" -c shutdown
Open On-Chip Debugger 0.11.0
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
debug_level: 2

Info : auto-selecting first available session transport "hla_swd". To override use 'transport select <transport>'.
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
Info : clock speed 1000 kHz
Info : STLINK V2J29S7 (API v2) VID:PID 0483:3748
Info : Target voltage: 2.932633
Error: Unable to set adapter speed
Error: init mode failed (unable to connect to the target)

make[1]: *** [mass_erase] Error 1
make: *** [factory_reset_21] Error 2

When I connected stlink-v2, the debug kits and the crazyflie, the crazyflie restarted, which may mean that the connection was successful. I'm not sure if this is a stlink-v2 issue. If this not the issue, then what else could it be?

**Comment by hmllr** [comment]
Hi :)
This seems to be an stlink issue - I don´t think the Crazyflie should restart (especially as this means the STM32 restarts, which you are not touching directly anyway) when you connect the debugger, but would anyway in general avoid plugging in and out connectors while the Crazyflie is turned on, to prevent accidental shortcuts which could potentially damage components. 
If the connection would be successful, it would write so in the terminal output - so it could help to double check that all your cables are properly connected to the nrf (following [https://www.bitcraze.io/products/debug-adapter-kit/#tab-id-11-3](url)).
Hope this helps,
Hanna

**Reply by shang-huan** [reply]
OK, I will try again. Thanks for your answer.
